src/components/Card.js

```diff
@@ -28,8 +36,18 @@ export default class Card {
     this._imageElement.alt = this._text;
   }
 
+  _setCardLike() {
+    if (this._isLiked){
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+
+  }
+
   _toggleLike() {
     // Toggle the like button class on click
+    this._handleCardLike(this._id, this._isLiked);
     this._likeButton.classList.toggle("card__like-button_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.